src/commands/upload_dart_symbol_map.rs

```diff
@@ -0,0 +1,165 @@
+    let mut ids: Vec<_> = dif.ids().into_iter().filter(|id| !id.is_nil()).collect();
+
+    // Ensure a single, unambiguous Debug ID
+    ids.sort();
+    ids.dedup();
+    match ids.len() {
+        0 => bail!(
+            "No debug identifier found in the provided debug file ({}). Ensure the file contains an embedded Debug ID.",
+            debug_file_path
+        ),
+        1 => {
+            let debug_id = ids.remove(0);
+
+            // Validate the dartsymbolmap JSON: must be a JSON array of strings with even length
+            let mapping_file_bytes = std::fs::read(mapping_path)
+                .with_context(|| format!("Failed to read mapping file at {}", mapping_path))?;
+            let mapping_entries: Vec<String> = serde_json::from_slice(&mapping_file_bytes)
+                .context("Invalid dartsymbolmap: expected a JSON array of strings")?;
+
+            if mapping_entries.len() % 2 != 0 {
+                bail!(
+                    "Invalid dartsymbolmap: expected an even number of entries (pairs), got {}",
+                    mapping_entries.len()
+                );
+            }
+
+            // Prepare upload object
+            let file_name = Path::new(mapping_path)
+                .file_name()
+                .and_then(OsStr::to_str)
+                .unwrap_or(mapping_path)
+                .to_string();
+
+            let mapping_len = mapping_file_bytes.len();
+            let object = DartSymbolMapObject {
+                bytes: mapping_file_bytes,
+                name: file_name.clone(),
+                debug_id,
+            };
+
+            // Prepare chunked upload
+            let api = Api::current();
+            let (org, project) = Config::current().get_org_and_project(matches)?;
+            let chunk_upload_options = api
+                .authenticated()?
+                .get_chunk_upload_options(&org)?
+                .ok_or_else(|| anyhow::anyhow!(
+                    "server does not support chunked uploading. Please update your Sentry server."
+                ))?;
+
+            if !chunk_upload_options.supports(ChunkUploadCapability::DartSymbolMap) {
+                bail!(
+                    "Server does not support uploading Dart symbol maps via chunked upload. Please update your Sentry server."
+                );
+            }
+
+            // Early file size check against server or default limits (same as debug files)
+            let effective_max_file_size = if chunk_upload_options.max_file_size > 0 {
+                chunk_upload_options.max_file_size
+            } else {
+                DEFAULT_MAX_DIF_SIZE
+            };
+
+            if (mapping_len as u64) > effective_max_file_size {
+                bail!(
+                    "The dartsymbolmap '{}' exceeds the maximum allowed size ({} bytes > {} bytes).",
+                    mapping_path,
+                    mapping_len,
+                    effective_max_file_size
+                );
+            }
```

Comment:
wasn't sure if we want this but added it to be safe

Reply:
I think it doesn't hurt, may help, so keep it :)